File: src/main/java/com/mikea/gae/rx/Observers.java
package com.mikea.gae.rx;

/**
 * @author mike.aizatsky@gmail.com
 */
public class Observers {
    public static <T> IObserver<T> asObserver(final IAction<T> action) {
        return new IObserver<T>() {
            @Override
            public void onCompleted() {
            }

            @Override
            public void onError(Exception e) {
                throw new UnsupportedOperationException(e);
            }

            @Override
            public void onNext(T value) {
                action.perform(value);
            }
        };
    }
}

File: src/main/scala/com/mikea/gae/rx/base/Observers.scala
package com.mikea.gae.rx.base

/**
 * @author mike.aizatsky@gmail.com
 */
object Observers {
  def asObserver[T](action: (T) => Unit): IObserver[T] = {
    new IObserver[T] {
      def onCompleted() {
      }

      def onError(e: Exception) {
        throw new UnsupportedOperationException(e)
      }

      def onNext(value: T) {
        action(value)
      }
    }
  }
}